### ISSUE TITLE
Violation of rule IDE0032 is not shown if property is declared virtual

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#40123]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/823058/violation-of-rule-ide0032-is-not-shown-if-property.html)._

---
<p>Visual Studio does not show de violation of the rule IDE0032 suggesting the use of auto-properties if the property using the private member is declared as virtual.</p><p>Here go both examples:</p><p>a) With virtual declaration</p><p><a href="https://developercommunity.visualstudio.com/storage/attachments/105259-capture1.jpg"> Image:105259-capture1.jpg</a></p><p><p>b) Without virtual declaration</p><p><a href="https://developercommunity.visualstudio.com/storage/attachments/105260-capture2.jpg"> Image:105260-capture2.jpg</a></p>

---
#### Visual Studio Feedback System on 11/18/2019, 00:04 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by CyrusNajmabadi** [comment]
Yes, this is by design.  The code says as much:

```c#
            // The property can't be virtual.  We don't know if it is overridden somewhere.  If it 
            // is, then calls to it may not actually assign to the field.
            if (property.IsVirtual || property.IsOverride || property.IsSealed)
            {
                return;
            }
```

**Comment by sharwell** [comment]
@CyrusNajmabadi this refactoring would not break code even if the property is virtual and overridden somewhere.

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi this refactoring would not break code even if the property is virtual and overridden somewhere.

How so?  Consdier something as simple as:

```c#
class A
{
    private int _id;
    public virtual int Id { get => _id; set => _id = value; }

    public int GetRawId() => _id;
    public void Whatever() => _id = 1;
}

class FixedId : A 
{
    public override int Id { get => -1 set { } }
}
```

This would definitely change behavior if you converted the above into:

```c#
class A
{
    public virtual int Id { get; set; }

    public int GetRawId() => Id;
    public void Whatever() => Id = 1;
}
```

**Comment by sharwell** [comment]
👍 So the diagnostic should be limited to cases where the only references to `_id` occur within `Id`.

**Comment by CyrusNajmabadi** [comment]
@sharwell Yes.  that would be another case we could look for! :)

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback.
